### ISSUE TITLE
Allow a passthrough of an AMI ID to the module for node groups

### ISSUE THREAD

**Issue by hhoover** [isovalent/terraform-aws-eks#6]
The module as-is will search for an image based on a provided `ami_name_filter` which works 99.99% of the time, unless you want to use a Bottlerocket AMI, only available in the SSM parameter store. One can perform a lookup of this in their downstream Terraform like so:

```
data "aws_ssm_parameter" "bottlerocket_image_id" {
  name = "/aws/service/bottlerocket/aws-k8s-${var.kubernetes_version}/x86_64/latest/image_id"
}

data "aws_ami" "bottlerocket_image" {
  owners = ["amazon"]
  filter {
    name   = "image-id"
    values = [data.aws_ssm_parameter.bottlerocket_image_id.value]
  }
}
```

then pass that into the EKS module as `data.aws_ami.bottlerocket_image.id`, however currently the module MUST do a filter lookup:

https://github.com/isovalent/terraform-aws-eks/blob/3914bbbcb0b3cd3ba024f4339b6ca8f267c3d6c8/eks.tf#L115

Desired behavior: make the AMI a conditional if the filter is provided, or a variable if it is not.

**Comment by hhoover** [comment]
I did eventually get a name filter to work, but it has a variable in it. `bottlerocket-aws-k8s-${var.kubernetes_version}-x86_64-*`

So one could pass a variable name filter for Ubuntu or something equally useless then in the module pass the bottlerocket name filter, but either way... changes need to be made to the module to allow for bottlerocket use:

https://github.com/terraform-aws-modules/terraform-aws-eks/blob/9022013844a61193a2f8764311fb679747807f5c/examples/bottlerocket/main.tf#L25-L52

**Comment by liyihuang** [comment]
this is implemented by 1.15.5 release
